### DISCUSSION TITLE
Data assimilation observation domain file

### DISCUSSION THREAD

**Discussion by cbachand** [NASA-LIS/LISF#1142] [Q&A]
Hi, I have been trying to use LIS to generate an ensemble by applying perturbations to the forcing variables. I was told by Sujay that perturbations are not applied correctly if DA is turned off. To overcome this problem, he recommended tricking LIS into thinking it is doing DA by providing a nonexistent observation directory, so that no data assimilation happens, but perturbations are still applied.

I have been trying to implement this, but am having trouble figuring out what to use for my "Data assimilation observation domain file". LIS stops running if this file is left unspecified. 

Can somebody explain to me the purpose of this file? I'm not actually applying DA, so will it affect my model run at all? 

I have tried using my lis_input.nc file, but I get the warning: "Error in nf90_get_att: GRIDTYPE" and the error: "ESMF_DistGridCreate failed: obsgridDG Stopping", which stops LIS from running.

Does anybody have suggestions on what I can use for this file so that LIS will run without stopping? 

For reference, my files are on discover: 

config file: /discover/nobackup/cbachand/INDUS_50km_merra2/lis.config_ensfinal_da
log files: /discover/nobackup/cbachand/INDUS_50km_merra2/log_ens_da/

**Comment by karsenau** [comment]
Hi @cbachand:

Typically your LDT-generated parameter-based lis_input.nc file should work for that entry.

Looking at your log files, I see the following error message:
 [INFO] Reading obs landmask map from lis_input.d01.nc
 **[INFO] Successfully read obs landmask map**
 [ERR] ESMF_DistGridCreate failed: obsgridDG Stopping.

It looks like the "obs landmask" map was successfully read in, so I am wondering if you could add some print statements within the routine where you see that above comment before any of the ESMF calls.  You could check if the target domain (which the interpolation to the gridspace that the EnKF steps will be computed on) was (1) successfully read in and (2) matches the landmask (and domain mask) that your parameters are on.

**Reply by karsenau** [reply]
Another recommendation that we wanted to offer is that the ESMF versions were updated for LIS and that may help address this error you are seeing.

I'm not sure which version of ESMF you are using, but please check and see about updating to the latest version, if it is older.  Check out the latest environment-based module for discover, in your LISF code:

env/discover/lisf_7_intel_2021.4.0

**Reply by cbachand** [reply]
Thank you all for the help. I am still struggling to get this working. I am using the most recent version of ESMF. I am not familiar with fortran and adding print statements to the source code has been challenging.

What is wrong with LIS' perturbations when DA is turned off? I get some ensemble spread, so some perturbations are being applied.
